### DISCUSSION TITLE
When use this analyzer?

### DISCUSSION THREAD

**Discussion by Magehernan** [microsoft/vs-threading#1112] [Q&A] [answered]
Hi, it is recommended use this analyser on any application? i am working on microservices and console application and when using a Lasy<Task<T>> it recomend using AsyncLazy... when i use it, it again tells me to use a JoiningTaskFactory about the UI thread.

So i start thinking that i am wrongly trying to use a analayzer for other types of application.

If i can use this on any type of application, what is the recommended way to get the JoiningTaskFactory? to make the analyzer stop warning me.

**Comment by AArnott** [comment, marked as answer]
Yes, most of these analyzers are useful for any type of application. If your application has no main thread, you can disable the specific rules that refer to it. For example the one about using JoinableTaskFactory can be turned off if you don't have a main thread.
